runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -50,89 +50,75 @@ class MemoryManager final {
     void erase(void *ptr) { _impl.erase(ptr); }
 };
 
-class Driver final {
+class ExecutionContext final {
```

Comment:
Really like this name change. I feel like it more accurately models this part of the runtime as the number of devices grow.